packages/sdk/src/types/runtime/userGroupCollection.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Context } from '.';
+
+// tslint:disable:no-bitwise
+
+/**
+ * A set of user group IDs. User groups are used to selectively enable several different
+ * properties of actors based on the memberships of the viewing user. All users not assigned
+ * a group are in the `default` group. See [[User.groups]], [[Appearance.enabled]].
+ */
+export default class UserGroupCollection extends Set<string> {
+    public static readonly ALL_PACKED = ~0;
+    public static readonly NONE_PACKED = 0;
+
+    private context: Context;
+
+    constructor(initialContents: Iterable<string> = null, context: Context = null) {
```

Comment:
It would be simpler to just always require a context passed to the constructor. Then you could delete setContext entirely.